File: Data Structures & Algorithms/remove-element/submission-3.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:

        l = 0
        n = len(nums)

        while l < n:
            print("nums[l]: " + str(nums[l]))
            if nums[l] == val:
                n -= 1
                nums[l] = nums[n]
                continue
            else:
                l += 1
        return n
        

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n = nums.length;
        int[] res = new int[n];

        res[0] = 1;
        for (int i = 1; i < n; i++){
            /* 
            i = 2
            res[2] = res[2-1] * nums[2-1]
            res[2] = res[1] * nums[1]
            res[2] = 1 * 2
            res[2] = 2

            i = 3
            res[3] = res[2] * nums[2]
            res[3] = 2 * 3
            res[3] = 6

            i = 4
            res[4] = res[3] * nums[3]
            res[4] = 6*4
            res[4] = 24
            */
            res[i] = res[i-1] * nums[i-1];
        }

        int postfix = 1;
        for (int i = n - 1; i >=0; i--) {
            res[i] *= postfix;
            postfix *= nums[i];
        }
        return res;
    }
}  
